### ISSUE TITLE
Solve circular rules with order priority

### ISSUE THREAD

**Issue by Specy** [eslint-stylistic/eslint-stylistic#779]
### Clear and concise description of the problem

I was trying to add eslint stylistic to my react codebase but i'm having issues with conflicting rules, especially inside jsx.

One example of this is applying the rule:
```js
{
    '@stylistic/jsx-closing-tag-location': ['error', 'line-aligned']
} 
```
and the indent rule of 4 spaces, the following code:
```tsx
{someCondition
            ? <Component
                    some="prop"
              >  <---- here
              </Component>
            : null //pretend there is other code here
}
```
the > will first say to apply `@stylistic/jsx-closing-tag-location` (how i want it to be) but then complain that it has the wrong number of indent spaces.

### Suggested solution

Allow to define rule order, so that when a cyclic rule is found (or in general when deciding which rules to apply and in which order), the rules will be applied in that order.

But since i'm guessing it's not possible to detect the order a priori, and instead "whatever rules" can be applied at a specific point are applied, the alternative is to run the rules, and when a cycle is found, stop at the higher order rule.

Let's say we have 3 rules, a, b, c. They are applied like:
```
a > b > c > b > c
```
In this case there is a cycle detected, we can add order importance to the rules and say that rule B has order of 1 and both `a` and `c` have the default order of 0, then when we are about to apply the last `c`, which would cause a cycle (which has lower order than the previous), we notice that `b` has higher order so `c` is not applied, end result would be:
```
a > b > c > b
```

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by Specy** [comment]
My issue specifically is related to https://github.com/eslint-stylistic/eslint-stylistic/issues/440
